CHANGELOG.md

```diff
@@ -1,5 +1,192 @@
 # Changelog
 
+## [1.28.0] - ?
+
+The new release of rustup has been a long time in the making and comes with substantial changes.
+
+The following improvements might require changes to how you use rustup:
+
+- rustup will no longer automatically install the active toolchain if it is not installed. [pr#3985]
+
+  * To ensure its installation, run `rustup toolchain install` with no arguments.
+  * The following command installs the active toolchain both before and after this change:
+    ```sh
+    rustup show active-toolchain || rustup toolchain install
+    # Or, on older versions of PowerShell:
+    rustup show active-toolchain; if ($LASTEXITCODE -ne 0) { rustup toolchain install }
+    ```
+
+- Installing a host-incompatible toolchain via `rustup toolchain install` or `rustup default` will
+  now be rejected unless you explicitly add the `--force-non-host` flag. [pr#4028]
+
+rustup now officially supports the following host platforms:
+
+  - `aarch64-pc-windows-msvc` [pr#3840]
+  - `loongarch64-unknown-linux-musl` [pr#3921]
+
+We have additionally made the following internal changes:
+
+- The default download backend has been changed from reqwest with native-tls to reqwest with rustls. [pr#3798]
+  - `RUSTUP_USE_CURL` and `RUSTUP_USE_RUSTLS` can still be used to change the download backend
+    if the new backend causes issues. If issues do happen, please [let us know](https://github.com/rust-lang/rustup/issues/3806).
```

Comment:
This might be a significant point. However, we(at least recent three releases) usually prefer placing all links at the end of the document. For example:
[let us know]: https://github.com/rust-lang/rustup/issues/3806

I simply followed the format used in previous changelog sections. That said, I think it's acceptable to have two different styles coexisting.

Reply:
> placing all links at the end of the document
I second this, despite the fact that this PR tries to maximally utilize the GitHub-generated detailed changelog. For the "manual" part, it'd still be better to simply follow the established convention.